js/app/cart/CartPortal.jsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import ReactDOM from 'react-dom';
+
+class CartPortal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.el = document.getElementById('cart-top');
```

Comment:
Just to make sure I understand: the « portal » is what manages the modal, right? 
If so, shouldn't the container be the `<body>` tag?

Reply:
no, the portal manage the cart in the top right of the page. It make sure it share the same state as the other cart when a client is on a restaurant page.

Reply:
Do we really need a portal for this?

Reply:
How would you do differently ? 

From what I understand, the portal allow you to render the element outside of the parent element tag. This is very interesting here because we want our TopCart element to share some state with the Cart element (total), but we want it to be rendered in the navigation bar.

You can find <a href=https://github.com/coopcycle/coopcycle-web/issues/95>here</a> more context about that issue.

Reply:
I think that the portal is a neat solution for our problem. Both carts (the one on the right and the top one) share the same state, so the same total.

Reply:
Ok, I see what you did there 🤓
I added some guidelines on how to achieve this in the main conversation.